src/manual/3rdparty/index.md

```diff
@@ -0,0 +1,74 @@
+
+*Do not* use clientIds for authentication.
+First, because mere posession of a credential with a clientId and some 44-character accessToken does not prove anything until you have validated that the accessToken is valid.
+Second, even if you validate the accessToken, TaskCluster is fairly permissive in creation of temporary credentials with arbitrary clientIds, by design.
+The information you may rely on for authorization is contained in the list of scopes returned from the [auth.authenticateHawk](/reference/platform/auth/api-docs#authenticateHawk) method.
+
+# Approaches
+
+## Frontend-Only Use
+
+If your application is a frontend-only tool, or only needs to call TaskCluster APIs on behalf of the user, you can perform the Access Grant process, then store the resulting credentials in the JavaScript heap or (to survive reloads) LocalStorage.
+Simply use those credentials along with the TaskCluster client to make calls to TaskCluster APIs.
+
+If you would like to perform additional error-checking, you can use those credentials to call [auth.currentScopes](/reference/platform/auth/api-docs#currentScopes).
+
+## Authorizing With Scopes
+
+The ideal way for your service to use TaskCluster credentials is to accept TaskCluster authentication on calls to your backend API.
+This is quite simple: call [auth.authenticateHawk](/reference/platform/auth/api-docs#authenticateHawk) from your backend with the appropriate parts of the HTTP request.
+Then verify that the returned scopes satisfy the scopes required for the operation being protected.
+There is no need to "register" the scopes you would like to use, but see the [namespaces document](/manual/devel/namespaces) for guidance on selecting appropriate names.
+
+The advantage of this approach is that it facilitates service re-use: anyone who is familiar with TaskCluster APIs can call your API, whether from a task, the command line, the browser, or another service.
+Furthermore, the backend never sees the credentials, just the Hawk signature.
+
+If you build a user interface around this approach, it is safe to display the clientId to the user so they can recognize the login.
+Just be cautious of the warning above regarding using clientIds for authentication.
+
+## Authenticating With Scopes
+
+If you need to know *who* the user is, such as to look up associated information in a backend database, do not use the provided clientId.
+Instead, treat the scopes returned from [auth.authenticateHawk](/reference/platform/auth/api-docs#authenticateHawk) as assertions of identity and group membership.
+
+In particular, look for scopes of the forms:
+
+ * `assume:mozilla-user:<email>`
+ * `assume:mozillians-user:<username>`
+ * `assume:mozilla-group:<group>`
+ * `assume:mozillians-group:<group>`
+
+This is one of the rare cases where you should *not* apply scope expansion; that is, posessing `assume:mozillians-user:*` should not allow a user to see data for all mozillians!
```

Comment:
For authorization, you should and do disagree.  For *authentication*, we do not want 3rd party services to accidentally grant us, the TC admins, godlike access to their services.